packages/trader/src/Modules/CFD/Containers/cfd-password-modal.tsx

```diff
@@ -453,24 +588,24 @@ const CFDServerModalWarning = ({ show_warning = true, platform }) => {
     );
 };
 
-const CFDServerForm = ({ ...props }) => {
+const CFDServerForm = ({ ...props }: TCFDServerFormProps) => {
     const available_servers = React.useMemo(() => {
         return [...props.mt5_trading_servers]
             .map(server => {
                 // Transform properties to support radiogroup
                 return {
                     ...server,
                     ...{
-                        label: `${server.geolocation.region} ${
-                            server.geolocation.sequence === 1 ? '' : server.geolocation.sequence
+                        label: `${server.geolocation?.region} ${
+                            server.geolocation?.sequence === 1 ? '' : server.geolocation?.sequence
                         } ${server.disabled ? `(${server.message_to_client})` : ''}`,
                         value: server.id,
                         disabled: server.disabled,
                     },
                 };
             })
-            .sort((a, b) => (a.recommended ? a : b))
-            .sort((a, b) => a.disabled - b.disabled);
+            .sort((a, _b) => (a.recommended ? -1 : 1))
```

Comment:
I had to fix this callback because the return value of the callback that is passed into the sort() method should be a number, not a sorted item. The way I implemented it is based on an assumption that the initial design was to make an item, which has a true 'recommended' value, go before the next item.